sdk/core/Azure.Core/src/ClientOptions.cs

```diff
@@ -86,22 +86,18 @@ public HttpPipelineTransport Transport
         /// <param name="position">The position of policy in the pipeline.</param>
         public void AddPolicy(HttpPipelinePolicy policy, HttpPipelinePosition position)
         {
-            switch (position)
+            if (position != HttpPipelinePosition.PerCall &&
+                position != HttpPipelinePosition.PerRetry &&
+                position != HttpPipelinePosition.BeforeTransport)
             {
-                case HttpPipelinePosition.PerCall:
-                    PerCallPolicies.Add(policy);
-                    break;
-                case HttpPipelinePosition.PerRetry:
-                    PerRetryPolicies.Add(policy);
-                    break;
-                default:
-                    throw new ArgumentOutOfRangeException(nameof(position), position, null);
+                throw new ArgumentOutOfRangeException(nameof(position), position, null);
             }
-        }
 
-        internal IList<HttpPipelinePolicy> PerCallPolicies { get; }
+            Policies ??= new();
+            Policies.Add((position, policy));
+        }
 
-        internal IList<HttpPipelinePolicy> PerRetryPolicies { get; }
+        internal List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)>? Policies { get; private set; }
```

Comment:
Are there any size issues with using tuples?  KVP might be cheaper?

Reply:
No issues AFAIK. Both are defined with 2 properties/fields.

Reply:
Yeah, sorry, these get turned into `ValueTuple<...>` which means no large ToString, etc.